index.rst

```diff
@@ -1,61 +1,173 @@
+We expect a maximum of about 10 million alerts per night, each about 80 kilobytes.
+LSE-163 :cite:`LSE-163` projects about 300 nights of observations per year.
+
+Combining these figures, we project 80 gigabytes of alert data per night, and 2.4 terabytes per year.
+
+Existing Requirements
+---------------------
+
+The database is referenced in a handful of existing project requirements documents:
+
+ - OSS-REQ-0128 "Alerts" :cite:`LSE-30`:
+
+     The Level 1 Data Products shall include the Alerts produced as part  of the nightly Alert Production.
+
+ - OSS-REQ-0185 "Transient Alert Query" :cite:`LSE-30`:
+
+     All published transient alerts, as well as all reprocessed historical alerts generated as part of a Data Release, shall be available for query.
+
+ - DMS-REQ-0094 "Keep Historical Alert Archive" :cite:`LSE-61`:
 
-.. TODO: Delete the note below before merging new content to the master branch.
+     The DMS shall preserve and keep in an accessible state an alert archive with all issued alerts for a historical record and for false alert analysis.
+
+ - DMS-PRTL-REQ-0033 "Queries on the Alerts Database" :cite:`LDM-554`:
+
+     The Portal aspect shall provide a query interface to the Alert Database, allowing searches based on parameters which shall include, but may not be limited to, Alert ID, time of alert, position on the sky, filter, and alert characteristics.
+
+ - DMS-PRTL-REQ-0048 "Alert Visualization" :cite:`LDM-554`:
+
+     The Portal aspect shall provide for the users a "property sheet" for the contents of an alert packet including, but not necessarily limited to, the alert postage stamp image, the postage stamp time series, the photometric time series, the source and object information (e.g., position, brightness).
+
+Proposed Implementation
+=======================
+
+We can satisfy these design inputs by storing serialized Avro alert data (the same bytes sent via Kafka to brokers) in a S3-like object store, indexed by a unique alert ID.
+Each alert packet corresponds to one object in the object store.
 
 .. note::
 
-   **This technote is not yet published.**
+   An alternative would be to combine many packets into a block in the object store, perhaps of about 100 alert packets.
+   This might permit more efficient storage and retrieval.
+   Storage might be more efficient because compression would be better when storing many alerts.
+   Retrieval might be more efficient because it might tune the outgoing flows into a smaller number of TCP connections which get a chance to grow window sizes.
 
-   Design document for a database of all alerts sent by the Rubin observatory.
+   But this would be more complex, and make writing more difficult, as writes need to append to existing data which would require coordination between writers, so this design sticks to a simpler structure.
+
+An object store is used because it is cheap, scales well to handle terabytes of data, and should support parallel retrieval reasonably well.
```

Comment:
I'd suggest saying a little more about what technical features the object store needs or doesn't.

Reply:
I haven't forgotten this - I can get to it when we feel confident that we have the rest of the doc roughly right.